pandas/tseries/tests/test_timedeltas.py

```diff
@@ -2052,6 +2052,22 @@ def test_add_overflow(self):
                   to_timedelta(['7 seconds', pd.NaT, '4 hours']))
         tm.assert_index_equal(result, exp)
 
+    def test_boolean_indexing(self):
+        # GH 14946
```

Comment:
put this in pandas/tests/indexing

create a new class for TimedeltaIndex

might be some other timedelta indexing routines you can group with it